### PR TITLE
Async

### PR DESCRIPTION
**Pull request by StorytellerCZ**
Continuation of #361

### CODE REVIEW

**Comment by StorytellerCZ** [comment]
Now we mainly need to add test suite for async.

File: .DS_Store
Comment (jankapunkt) [inline .DS_Store:1]: I think we should remove .DS_Store files

File: someapp/packages/roles
Comment (jankapunkt) [inline someapp/packages/roles:1]: is this a softlink file?

**Review by jankapunkt — CHANGES_REQUESTED** [review]
somehow we ended up with two meteor projects for testing, `someapp` and `testapp` and I think `someapp` has accidentally been added.

File: someapp/packages/roles
Comment (StorytellerCZ) [reply someapp/packages/roles:1]: Removed someapp.

File: .DS_Store
Comment (StorytellerCZ) [reply .DS_Store:1]: Removed and added to `.gitignore`

**Review by jankapunkt — APPROVED** [review]

**Comment by StorytellerCZ** [comment]
OK, figured out the tests, now it is just about converting the test suite to it.

**Comment by StorytellerCZ** [comment]
Got all the async tests running.
I have removed migration tests which I don't feel need to be in the async tests as they should have been run before upgrading to Meteor 3, maybe we should mention that in the readme.

There seems to be a bug in the async version of detecting roles cycles in `_addRoleToParentAsync` function, which seem to break these tests. Right now I don't see what it could be as the async transition seems to have been done well. Fixing that I think could fix at least half the failing tests.

**Comment by bratelefant** [comment]
> Got all the async tests running. I have removed migration tests which I don't feel need to be in the async tests as they should have been run before upgrading to Meteor 3, maybe we should mention that in the readme.
> 
> There seems to be a bug in the async version of detecting roles cycles in `_addRoleToParentAsync` function, which seem to break these tests. Right now I don't see what it could be as the async transition seems to have been done well. Fixing that I think could fix at least half the failing tests. @bratelefant any ideas?

Guess I fixed this, cf #380 

Sometimes it's the small things that matter ;)

**Comment by jankapunkt** [comment]
Yup, tests are fixed, thanks @bratelefant 🙌

@StorytellerCZ from my end it's good, should we do a final review or do you think it's fine?

**Comment by StorytellerCZ** [comment]
I think we might need client side tests for async as well, but I think it is fine. I will prepare a feature release once I get some time.

**Comment by StorytellerCZ** [comment]
@jankapunkt so the client tests were much easier than I thought, honestly I kind of find them lacking, but that is for another time. For me we can merge and release this as a new feature version. I will have to look into re-generating the docs with this.

**Review by jankapunkt — APPROVED** [review]
Ltgm

**Pull request merged**